### PR TITLE
Add quality declaration spdlog_vendor

### PR DESCRIPTION
**Pull request by Blast545**
This is a quality declaration for spdlog_vendor as described in the proposed REP-2004 (see ros-infrastructure/rep#218).

Checklist in #5 can be checked first to sort out missing parts of the declaration.

Signed-off-by: Jorge Perez <jjperez@ekumenlabs.com>

### CODE REVIEW

**Comment by Blast545** [comment]
Before merging, I will check why copyright test result is not shown in [CI link](https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_release/lastBuild/testReport/spdlog_vendor/) (seems to be included in the `package.xml`) file.

File: Quality_Declaration.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+`spdlog_vendor` does not have an API of its own, and will limit its dependency on `spdlog` to API stable releases within a released ROS distribution.
+
+### ABI Stability Policy [1.v]
+
+`spdlog_vendor` does not have an ABI of its own, and will limit its dependency on `spdlog` to ABI stable releases within a released ROS distribution.
+
+### ABI and ABI Stability Within a Released ROS Distribution [1.vi]
+
+As `spdlog_vendor` does not come with its own API/ABI, changes will not affect API/ABI stability.
```
Comment (brawner) [inline Quality_Declaration.md]:
```suggestion
As `spdlog_vendor` does not have its own API/ABI, changes will not affect API/ABI stability.
```

File: Quality_Declaration.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+`spdlog_vendor` does not currently document its features.
+
+### Public API Documentation [3.ii]
+
+`spdlog_vendor` does not have an API and therefore does not require API documentation.
+
+### License [3.iii]
+
+The license for `spdlog_vendor` is Apache 2.0, and a summary is in each source file, the type is declared in the `package.xml` manifest file, and a full copy of the license is in the `LICENSE` file. The vendored library, `spdlog` license is MIT as stated in its quality declaration document (Section 5.iii).
```
Comment (brawner) [inline Quality_Declaration.md]: Link to package.xml, LICENSE file and the external QD.

File: Quality_Declaration.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+The copyright holders each provide a statement of copyright in each source code file in `spdlog_vendor`.
+
+There is an automated test which runs a linter that ensures each file has at least one copyright statement. Latest linter result report can be seen [here](https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_release/lastBuild/testReport/spdlog_vendor/copyright/).
+
+## Testing [4]
+
+### Feature Testing [4.i]
+
+`spdlog_vendor` does not have feature testing. For vendored packages, this will be considered to be addressed in the external dependency.
```
Comment (brawner) [inline Quality_Declaration.md]: I don't think 4.i - 4.iv are needed here. I would just under section 4, this package is simply a vendor package and does not need testing of its own.

File: Quality_Declaration.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+## Dependencies [5]
+
+### Direct and Optional Runtime ROS Dependencies [5.i]/[5.ii]
+
+`spdlog_vendor` does not have direct/optional runtime ROS dependencies.
+
+### Direct Runtime non-ROS Dependency [5.iii]
+
+`spdlog_vendor` depends directly on the external dependency `spdlog`, which is qualified as quality level 4 in its [Quality Declaration](./spdlog_Q_DECLARATION.md).
```
Comment (brawner) [inline Quality_Declaration.md]: Link will be wrong for external QD. https://github.com/ros2/spdlog_vendor/pull/8

File: Quality_Declaration.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+|6.i| Support targets Tier1 ROS platforms| ✓ |
+|7| Security | --- |
+|7.i| Vulnerability Disclosure Policy | ☓ |
+
+Comparing this table with the [Quality Level Comparison Chart of REP2004](https://github.com/ros-infrastructure/rep/blob/master/rep-2004.rst#quality-level-comparison-chart) lead us to decide that this package qualifies to Quality Level 4.
+
+To reach Quality level 3, adding a Vulnerability Disclosure will be needed.
+To reach Quality level 2, adding features documentation will be needed.
+
+non-ROS dependencies Quality level will need to be adjusted accordignly to be equal or below to claims made in this Quality Declaration.
```
Comment (brawner) [inline Quality_Declaration.md]:
```suggestion
non-ROS dependencies Quality level will need to be adjusted accordingly to be equal or below to claims made in this Quality Declaration.
```

What are you trying to say here?

Comment (Blast545) [reply Quality_Declaration.md]: I was trying to imply that to have upper levels on this QD, the `spdlog` external dependency have to increase its QL as well as an extra thought

Comment (brawner) [reply Quality_Declaration.md]: How about:

"spdlog_vendor is limited by the quality level of the upstream dependency spdlog. In order to reach higher quality levels, the issues limiting spdlog will either need to be addressed by spdlog itself, or through adequate quality control measures in this package."

File: QUALITY_DECLARATION.md [hunk 6]
```diff
@@ -70,7 +70,7 @@ All pull requests must resolve related documentation changes before merging.
 
 ### License [3.iii]
 
-The license for `spdlog_vendor` is Apache 2.0, and a summary is in each source file, the type is declared in the `package.xml` manifest file, and a full copy of the license is in the `LICENSE` file. The vendored library, `spdlog` license is MIT as stated in its quality declaration document (Section 5.iii).
+The license for `spdlog_vendor` is Apache 2.0, and a summary is in each source file, the type is declared in the [`package.xml`](./package.xml) manifest file, and a full copy of the license is in the [LICENSE](./LICENSE.md) file. The vendored library, `spdlog` license is MIT as stated in its [Quality declaration](./SPDLOG_QUALITY_DECLARATION.md) document (Section 5.iii).
```
Comment (brawner) [inline QUALITY_DECLARATION.md]: Fix license file link.

File: QUALITY_DECLARATION.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+`spdlog_vendor` follows the recommended guidelines for ROS Core packages in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change-control-process).
+
+### Contributor Origin [2.ii]
+
+This package uses DCO as its confirmation of contributor origin policy. More information can be found in [CONTRIBUTING](./CONTRIBUTING.md).
+
+### Peer Review Policy [2.iii]
+
+All pull requests will be peer-reviewed, check [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change-control-process) for additional information.
```
Comment (brawner) [inline QUALITY_DECLARATION.md]:
```suggestion
All pull requests will be peer-reviewed, check the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change-control-process) for additional information.
```

File: QUALITY_DECLARATION.md [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+
+`spdlog_vendor` does not have an API of its own, and will limit its dependency on `spdlog` to API stable releases within a released ROS distribution.
+
+### ABI Stability Policy [1.v]
+
+`spdlog_vendor` does not have an ABI of its own, and will limit its dependency on `spdlog` to ABI stable releases within a released ROS distribution.
+
+### ABI and ABI Stability Within a Released ROS Distribution [1.vi]
+
+As `spdlog_vendor` does not have its own API/ABI, changes will not affect API/ABI stability.
```
Comment (brawner) [inline QUALITY_DECLARATION.md]:
```suggestion
`spdlog_vendor` will limit its dependency on `spdlog` to API and ABI stable releases within a released ROS distribution.
```

File: QUALITY_DECLARATION.md [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+
+### Documentation Policy [2.v]
+
+All pull requests must resolve related documentation changes before merging.
+
+## Documentation [3]
+
+### Feature Documentation [3.i]
+
+`spdlog_vendor` does not currently document its features.
```
Comment (brawner) [inline QUALITY_DECLARATION.md]: Same comments from the libyaml_vendor review.

File: QUALITY_DECLARATION.md [hunk 10]
```diff
@@ -20,17 +20,13 @@ Below are the rationales, notes, and caveats for this claim, organized by each r
 
 This is a vendor package for `spdlog` and as such does not declare its own API.
 
-### API Stability Policy [1.iv]
+### API Stability Policy [1.iv]/[1.vi]
 
-`spdlog_vendor` does not have an API of its own, and will limit its dependency on `spdlog` to API stable releases within a released ROS distribution.
+`spdlog_vendor` does not have an API of its own, and will limit its dependency on `libyaml` to API stable releases within a released ROS distribution.
```
Comment (brawner) [inline QUALITY_DECLARATION.md]:
```suggestion
`spdlog_vendor` does not have an API of its own, and will limit its dependency on `spdlog` to API stable releases within a released ROS distribution.
```

File: QUALITY_DECLARATION.md [hunk 11]
```diff
@@ -20,17 +20,13 @@ Below are the rationales, notes, and caveats for this claim, organized by each r
...
 
-### ABI Stability Policy [1.v]
+### ABI Stability Policy [1.v]/[1.vi]
 
-`spdlog_vendor` does not have an ABI of its own, and will limit its dependency on `spdlog` to ABI stable releases within a released ROS distribution.
-
-### ABI and ABI Stability Within a Released ROS Distribution [1.vi]
-
-As `spdlog_vendor` does not have its own API/ABI, changes will not affect API/ABI stability.
+`spdlog_vendor` does not have an ABI of its own, and will limit its dependency on `libyaml` to ABI stable releases within a released ROS distribution.
```
Comment (brawner) [inline QUALITY_DECLARATION.md]:
```suggestion
`spdlog_vendor` does not have an ABI of its own, and will limit its dependency on `spdlog` to ABI stable releases within a released ROS distribution.
```

File: QUALITY_DECLARATION.md [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+
+This package uses DCO as its confirmation of contributor origin policy. More information can be found in [CONTRIBUTING](./CONTRIBUTING.md).
+
+### Peer Review Policy [2.iii]
+
+All pull requests will be peer-reviewed, check the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change-control-process) for additional information.
+
+### Continuous Integration [2.iv]
+
+All pull request must pass CI on all [tier 1 platforms](https://www.ros.org/reps/rep-2000.html#support-tiers).
```
Comment (brawner) [inline QUALITY_DECLARATION.md]:
```suggestion
All pull requests must pass CI on all [tier 1 platforms](https://www.ros.org/reps/rep-2000.html#support-tiers).
```

**Review by brawner — APPROVED** [review]
Fix the copy and paste errors, otherwise looks good.

**Pull request merged**